src/ESPEasyEth.ino

```diff
@@ -0,0 +1,35 @@
+
+#ifdef HAS_ETHERNET
+  #define ETH_CLK_MODE ETH_CLOCK_GPIO17_OUT
+  #define ETH_PHY_POWER 12
+  #include <ETH.h>
+
+
+String EthGetHostname()
+{
+  String hostnameToReturn(Settings.getHostname());
+  hostnameToReturn.replace(" ", "-");
+  hostnameToReturn.replace("_", "-"); // See RFC952
+  return hostnameToReturn;
+}
+
+bool prepareEth() {
+  char hostname[40];
+  safe_strncpy(hostname, EthGetHostname(), sizeof(hostname));
+  ETH.setHostname(hostname);
```

Comment:
Why not use `ETH.setHostname(EthGetHostname().c_str());` ?

Reply:
:D 
Right, idiotic....